benches/end2end.rs

```diff
@@ -0,0 +1,386 @@
+use blstrs::Scalar as Fr;
+use criterion::{
+    black_box, criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion, SamplingMode,
+};
+use lurk::{
+    eval::{
+        empty_sym_env,
+        lang::{Coproc, Lang},
+        Evaluator,
+    },
+    field::LurkField,
+    proof::nova::NovaProver,
+    proof::Prover,
+    ptr::Ptr,
+    public_parameters,
+    store::Store,
+};
+use std::sync::Arc;
+use std::time::Duration;
+
+const DEFAULT_REDUCTION_COUNT: usize = 10;
+fn go_base<F: LurkField>(store: &mut Store<F>, a: u64, b: u64) -> Ptr<F> {
+    let program = format!(
+        r#"
+(let ((foo (lambda (a b)
+              (letrec ((aux (lambda (i a x)
+                               (if (= i b)
+                                     x
+                                     (let ((x (+ x a))
+                                            (a (+ a (* b 2))))
+                                       (aux (+ i 1) a x))))))
+                       (let ((x (+ (* a b) 4)))
+                         (aux 0 a x))))))
+  (foo {a} {b}))
+"#
+    );
+
+    store.read(&program).unwrap()
+}
+
+fn end2end_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("end2end_benchmark");
+    group
+        .sampling_mode(SamplingMode::Flat)
+        .measurement_time(Duration::from_secs(120))
+        .sample_size(10);
+
+    let limit = 1_000_000_000;
+    let lang_vesta = Lang::<pasta_curves::Fq, Coproc<pasta_curves::Fq>>::new();
+    let lang_vesta_rc = Arc::new(lang_vesta.clone());
```

Comment:
This should be `lang_pallas`, right? Looking at the `eval` benchmark, it looks like maybe `Fp` was used and (wrongly?) designated as `lang_pallas`. In other words, I think `Fq` is the scalar field of Pallas.

Reply:
Whoops, reading on, looks like that was copied here too.

Reply:
The inverse comments therefore apply below. Maybe I'm confused, though.

Reply:
Yeah, I just copied from `eval`. Looking at pasta curves docs I thnk `Fq` is indeed the scalar field of Pallas -- will swap things around in another PR